homeassistant/components/samsungtv/media_player.py

```diff
@@ -60,54 +62,70 @@
 )
 
 
+async def _get_device_bridge(data):
+    """Get device bridge."""
+    return SamsungTVBridge.get_bridge(
+        data[CONF_METHOD],
+        data[CONF_HOST],
+        data[CONF_PORT],
+        data.get(CONF_TOKEN),
+    )
+
+
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up the Samsung TV from a config entry."""
-    ip_address = config_entry.data[CONF_IP_ADDRESS]
+    host = config_entry.data[CONF_HOST]
     on_script = None
     if (
         DOMAIN in hass.data
-        and ip_address in hass.data[DOMAIN]
-        and CONF_ON_ACTION in hass.data[DOMAIN][ip_address]
-        and hass.data[DOMAIN][ip_address][CONF_ON_ACTION]
+        and host in hass.data[DOMAIN]
+        and CONF_ON_ACTION in hass.data[DOMAIN][host]
+        and hass.data[DOMAIN][host][CONF_ON_ACTION]
     ):
-        turn_on_action = hass.data[DOMAIN][ip_address][CONF_ON_ACTION]
+        turn_on_action = hass.data[DOMAIN][host][CONF_ON_ACTION]
         on_script = Script(
             hass, turn_on_action, config_entry.data.get(CONF_NAME, DEFAULT_NAME), DOMAIN
         )
 
     # Initialize bridge
     data = config_entry.data.copy()
-    bridge = SamsungTVBridge.get_bridge(
-        data[CONF_METHOD],
-        data[CONF_HOST],
-        data[CONF_PORT],
-        data.get(CONF_TOKEN),
-    )
+    bridge = await _get_device_bridge(data)
     if bridge.port is None and bridge.default_port is not None:
         # For backward compat, set default port for websocket tv
         data[CONF_PORT] = bridge.default_port
         hass.config_entries.async_update_entry(config_entry, data=data)
-        bridge = SamsungTVBridge.get_bridge(
-            data[CONF_METHOD],
-            data[CONF_HOST],
-            data[CONF_PORT],
-            data.get(CONF_TOKEN),
-        )
+        bridge = await _get_device_bridge(data)
```

Comment:
Looks like this is created in two places

Reply:
Even three:

media_player.py:92:    bridge = await _get_device_bridge(data)
media_player.py:97:        bridge = await _get_device_bridge(data)
media_player.py:113:    bridge = await _get_device_bridge(data)

Reply:
And there isn't anything being awaited in `_get_device_bridge` so it doesn't need to be a co-routinue.  Rename to `_async_get_device_bridge`, remove `async` from `def`, and add `@callback`

Reply:
Done